packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -143,6 +144,10 @@ class SemanticsData extends Diagnosticable {
   /// [increasedValue], and [decreasedValue].
   final TextDirection textDirection;
 
+  /// The index indicating the order in which this node should
```

Comment:
I still feel a little uncomfortable about sending over an index for a global order. When you have a parent node with a 100 child nodes all specifying an order and you add a node at the very beginning, we will have to send all 100 child nodes over again, decode them on the engine side, translate them into platform-specific nodes, etc (we would even tell the OS that these 100 nodes have changed even though they didn't really).

The only thing that has changed is that after the parent node we now need to traverse to the new node instead of the old node 1.

I'd much prefer to just send that minimal diff. Or we should create a benchmark verifying that doing the ordering like this doesn't have a performance impact (the benchmark could maybe measure the performance of adding a child node to an existing list of n child nodes with and without an order set).

Reply:
OK, done.  I now push down the nextNodeId instead of the traversal index, and only mark those nodes where the next node has changed as dirty.  I'm not sure what impact this will have on implementing it for iOS yet, however.